lib/adapters/http.js

```diff
@@ -380,9 +380,9 @@ export default function httpAdapter(config) {
 
       // if decompress disabled we should not decompress
       if (config.decompress !== false) {
-        // if no content, but headers still say that it is encoded,
+        // if no content or HEAD request, but headers still say that it is encoded,
         // remove the header not confuse downstream operations
-        if (data && data.length === 0 && res.headers['content-encoding']) {
+        if ((method === 'HEAD' || data && data.length === 0) && res.headers['content-encoding']) {
```

Comment:
I assume that `data` is `undefined` in the scenario? Would it not be better to just check that then?

```
(typeof data === 'undefined' || data && data.length === 0)
```

Feels safer than making specific checks against method?

Reply:
The problem is this `data` is always undefined for `GET` requests and would also skip the all decompression on those.
As far as I could comprehend, `data` is only set before hand if it is a `POST` request.
The response stream is handled after this `data` variable is checked. If you would include this `undefined` check the unit tests 
 - should support gunzip error handling
 - should support transparent gunzip
will fail as they would not decompress the response anymore.

Reply:
I close this issue, if you have further questions just write a comment again.